### PR TITLE
feat: introduce storage node configuration

### PR DESCRIPTION
**Pull request by xx01cyx**
We can run the storage node as follows:

```
cargo run -- --data-store memdisk --disk-cache-size 1000000 --mem-cache-size 1000 --cache-path "my-cache" --max-disk-reader-buffer-size 10000000
```

See `config.rs` for more configurable fields.

### CODE REVIEW

File: storage-node/src/bin/storage_node.rs [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 #[tokio::main]
 async fn main() {
-    let _ = env_logger::builder()
+    // Init log.
+    if let Err(e) = env_logger::builder()
         .filter_level(log::LevelFilter::Info)
-        .try_init();
+        .try_init()
+    {
+        println!("Failed to init logger: {:?}", e);
```
Comment (unw9527) [inline storage-node/src/bin/storage_node.rs:12]: Why not log?

File: storage-node/src/server.rs [hunk 2]
```diff
@@ -105,12 +97,147 @@ pub async fn storage_node_serve(ip_addr: &str, port: u16) -> ParpulseResult<()>
     let routes = route.or(heartbeat).or(catch_all);
     let ip_addr: IpAddr = ip_addr.parse().unwrap();
     warp::serve(routes).run((ip_addr, port)).await;
+}
+
+pub async fn storage_node_serve(
```
Comment (unw9527) [inline storage-node/src/server.rs:98]: Do we have to explicitly match xxx to LRU or LRU-K for every type of data store...?

File: storage-node/src/storage_manager.rs [hunk 3]
```diff
@@ -5,23 +5,32 @@ use crate::{
...
 };
 
+use async_trait::async_trait;
 use bytes::Bytes;
 use log::debug;
 use parpulse_client::RequestParams;
 use tokio::sync::mpsc::Receiver;
 
+#[async_trait]
+pub trait StorageManager: Send + Sync {
```
Comment (unw9527) [inline storage-node/src/storage_manager.rs:14]: Why do we need to extract this out?

**Review by lanlou1554 — APPROVED** [review]
Do we need to also make `MAX_DISK_READER_BUFFER_SIZE` configurable?
Also, for the client parameters, I think we need to sync with cache2 team!
Others LGTM!

File: storage-node/src/bin/storage_node.rs (hunk 1 above)
Comment (xx01cyx) [reply storage-node/src/bin/storage_node.rs:12]: because log init fails

File: storage-node/src/server.rs (hunk 2 above)
Comment (xx01cyx) [reply storage-node/src/server.rs:98]: The concrete types have to match when we initialize a variable. We can't assign a variable as LRU or LRUK at the same time. So we have to do this.

File: storage-node/src/storage_manager.rs (hunk 3 above)
Comment (xx01cyx) [reply storage-node/src/storage_manager.rs:14]: to enable `route` to accept `StorageManager<C>` where C could be different

File: storage-node/src/server.rs (hunk 2 above)
Comment (unw9527) [reply storage-node/src/server.rs:98]: No. I mean if it is possible to modularize repetitive code.

**Review by lanlou1554 — APPROVED** [review]

**Pull request merged**